example/advection/advection_package.cpp

```diff
@@ -197,17 +210,26 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
   // add callbacks for HST output identified by the `hist_param_key`
   pkg->AddParam<>(parthenon::hist_param_key, hst_vars);
 
-  pkg->FillDerivedBlock = SquareIt;
+  if (fill_derived) {
+    pkg->FillDerivedBlock = SquareIt;
+  }
   pkg->CheckRefinementBlock = CheckRefinement;
   pkg->EstimateTimestepBlock = EstimateTimestepBlock;
 
   return pkg;
 }
 
 AmrTag CheckRefinement(MeshBlockData<Real> *rc) {
-  auto pmb = rc->GetBlockPointer();
   // refine on advected, for example.  could also be a derived quantity
-  auto v = rc->Get("advected").data;
+  auto pmb = rc->GetBlockPointer();
+  auto pkg = pmb->packages.Get("advection_package");
+  int num_vars = pkg->Param<int>("num_vars");
+  std::vector<std::string> vars;
+  for (int var = 0; var < num_vars; ++var) {
+    vars.push_back("advected_" + std::to_string(var));
+  }
+  // type is parthenon::VariablePack<CellVariable<Real>>
+  auto v = rc->PackVariables(vars); // Get("advected").data;
```

Comment:
```suggestion
  auto v = rc->PackVariables(vars);
```
We can probably remove the old commented out code.